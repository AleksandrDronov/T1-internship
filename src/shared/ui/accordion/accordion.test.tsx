import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Accordion } from ".";

describe("Accordion", () => {
  test("Accordion renders correctly", () => {
    const tree = renderer
      .create(
        <Accordion
          items={[
            { title: "Item 1", content: "Content 1" },
            { title: "Item 2", content: "Content 2" },
          ]}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("AccordionItem opens and closes", () => {
    render(<Accordion items={[{ title: "Item 1", content: "Content 1" }]} />);
    const button = screen.getByRole("button");
    fireEvent.click(button);
    expect(button).toHaveAccessibleName("Open accordion section");
    fireEvent.click(button);
    expect(button).toHaveAccessibleName("Close accordion section");
  });
});
