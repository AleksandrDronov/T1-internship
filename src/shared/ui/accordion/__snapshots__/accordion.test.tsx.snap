// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Accordion renders correctly 1`] = `
<ul
  className="accordion"
>
  <li
    className="item"
  >
    <button
      aria-expanded={false}
      aria-label="Close accordion section"
      className="button button tertiary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <div
        className="header"
      >
        <p
          className="text b12-web black left"
        >
          Item 1
        </p>
        <img
          alt=""
          src=""
        />
      </div>
      <p
        className="content text b13-web black left"
        style={
          {
            "height": 0,
          }
        }
      >
        <span
          className="body"
        >
          Content 1
        </span>
      </p>
    </button>
  </li>
  <li
    className="item"
  >
    <button
      aria-expanded={false}
      aria-label="Close accordion section"
      className="button button tertiary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <div
        className="header"
      >
        <p
          className="text b12-web black left"
        >
          Item 2
        </p>
        <img
          alt=""
          src=""
        />
      </div>
      <p
        className="content text b13-web black left"
        style={
          {
            "height": 0,
          }
        }
      >
        <span
          className="body"
        >
          Content 2
        </span>
      </p>
    </button>
  </li>
</ul>
`;
